From ef10d201870b630b2d148b1a0e351adb53bcaf47 Mon Sep 17 00:00:00 2001
From: "Darryl L. Pierce" <dpierce@redhat.com>
Date: Wed, 7 May 2014 07:42:15 -0400
Subject: [PATCH] NO-JIRA: Bumped max Proton version to 0.7

---
 qpid/cpp/src/amqp.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qpid/cpp/src/amqp.cmake b/qpid/cpp/src/amqp.cmake
index 14b660d..b4870bd 100644
--- a/qpid/cpp/src/amqp.cmake
+++ b/qpid/cpp/src/amqp.cmake
@@ -48,7 +48,7 @@ endif()
 
 set (amqp_default ${amqp_force})
 set (minimum_version 0.5)
-set (maximum_version 0.6)
+set (maximum_version 0.7)
 if (PROTON_FOUND)
     if (PROTON_VERSION LESS ${minimum_version})
         message(STATUS "Qpid proton ${PROTON_VERSION} is too old, require ${minimum_version} - ${maximum_version}; amqp 1.0 support not enabled")
-- 
1.9.0

