From 87125fc0e2ace2cfb9fb63a1fcf715022fa900cf Mon Sep 17 00:00:00 2001
From: "Darryl L. Pierce" <dpierce@redhat.com>
Date: Thu, 26 Jul 2012 17:18:27 -0400
Subject: [PATCH] Fixed db4 on Fedora 18.

---
 configure.ac             | 8 ++++----
 lib/MessageStoreImpl.cpp | 2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 4ed71dd..3103f37 100644
--- a/configure.ac
+++ b/configure.ac
@@ -181,7 +181,7 @@ LIBS=$gl_saved_libs
 
 # Require libdb_cxx (any version between 4.2 and 4.8), for the library, and for db_cxx.h.
 db4_devel_fail=0
-AC_CHECK_HEADER([db_cxx.h], ,[db4_devel_fail=1])
+AC_CHECK_HEADER([libdb4/db_cxx.h], ,[db4_devel_fail=1])
 test $db4_devel_fail == 1 && \
   AC_MSG_ERROR([db4-devel package missing. Please ensure both db4 and db4-devel are installed. (hint: "yum install db4-devel" should do it...)])
 
@@ -194,9 +194,9 @@ AC_SUBST([LIB_BERKELEY_DB])
 LIBS=$gl_saved_libs
 
 # Determine how to include db_cxx.h:
-# Red Hat needs <db4/db_cxx.h>, Debian needs <db_cxx.h>.
-AC_CHECK_HEADER([db4/db_cxx.h],
-                [DB_CXX_HEADER_PREFIX=db4/])
+# Red Hat needs <libdb4/db_cxx.h>, Debian needs <db_cxx.h>.
+AC_CHECK_HEADER([libdb4/db_cxx.h],
+                [DB_CXX_HEADER_PREFIX=libdb4/])
 if test x$DB_CXX_HEADER_PREFIX = x; then
   AC_CHECK_HEADER([db_cxx.h],
 		  [DB_CXX_HEADER_PREFIX=])
diff --git a/lib/MessageStoreImpl.cpp b/lib/MessageStoreImpl.cpp
index d8cd611..4545573 100644
--- a/lib/MessageStoreImpl.cpp
+++ b/lib/MessageStoreImpl.cpp
@@ -32,7 +32,7 @@
 #include "qmf/com/redhat/rhm/store/Package.h"
 #include "StoreException.h"
 #include <dirent.h>
-#include <db.h>
+#include <libdb4/db.h>
 
 #define MAX_AIO_SLEEPS 100000 // tot: ~1 sec
 #define AIO_SLEEP_TIME_US  10 // 0.01 ms
-- 
1.7.11.2

