--- cpp/src/Makefile.am	2010-10-27 14:01:27.000000000 -0400
+++ cpp/src/Makefile.am	2011-02-07 15:35:30.962680422 -0500
@@ -122,6 +122,8 @@
 tmoduledir = $(libdir)/qpid/tests
 tmodule_LTLIBRARIES=
 
+AM_CXXFLAGS += -DBOOST_FILESYSTEM_VERSION=2
+
 ## Automake macros to build libraries and executables.
 qpidd_CXXFLAGS = $(AM_CXXFLAGS) -DQPIDD_MODULE_DIR=\"$(dmoduledir)\" -DQPIDD_CONF_FILE=\"$(sysconfdir)/qpidd.conf\"
 libqpidclient_la_CXXFLAGS = $(AM_CXXFLAGS) -DQPIDC_MODULE_DIR=\"$(cmoduledir)\" -DQPIDC_CONF_FILE=\"$(confdir)/qpidc.conf\"
