diff -up qpid-0.8/cpp/src/qmf/SchemaMethod.cpp.size_t qpid-0.8/cpp/src/qmf/SchemaMethod.cpp
--- qpid-0.8/cpp/src/qmf/SchemaMethod.cpp.size_t	2011-01-12 09:06:36.842754605 -0500
+++ qpid-0.8/cpp/src/qmf/SchemaMethod.cpp	2011-01-12 09:07:00.802754933 -0500
@@ -163,7 +163,7 @@ void SchemaMethodImpl::encodeV1(qpid::ma
     Variant::Map map;
 
     map["name"] = name;
-    map["argCount"] = arguments.size();
+    map["argCount"] = (uint64_t)arguments.size();
     if (!desc.empty())
         map["desc"] = desc;
 
