diff -up qpid-0.14/cpp/src/qpid/broker/MessageGroupManager.cpp.size_t qpid-0.14/cpp/src/qpid/broker/MessageGroupManager.cpp
--- qpid-0.14/cpp/src/qpid/broker/MessageGroupManager.cpp.size_t	2012-04-17 10:59:20.000000000 +0200
+++ qpid-0.14/cpp/src/qpid/broker/MessageGroupManager.cpp	2012-04-17 10:59:40.000000000 +0200
@@ -280,7 +280,7 @@ void MessageGroupManager::query(qpid::ty
          g != messageGroups.end(); ++g) {
         qpid::types::Variant::Map info;
         info[GROUP_ID_KEY] = g->first;
-        info[GROUP_MSG_COUNT] = g->second.members.size();
+        info[GROUP_MSG_COUNT] = (uint64_t)g->second.members.size();
         info[GROUP_TIMESTAMP] = 0;   /** @todo KAG - NEED HEAD MSG TIMESTAMP */
         info[GROUP_CONSUMER] = g->second.owner;
         groups.push_back(info);
