--- cpp/configure.ac	2010-07-26 13:45:55.158378547 -0400
+++ cpp/configure.ac	2010-07-26 14:00:02.937354780 -0400
@@ -144,7 +144,7 @@
 AC_CHECK_PROG([RUBY], [ruby], [ruby])
 
 # Swig binding generator is needed for the script (Ruby, Python, etc.) bindings.
-AC_PROG_SWIG(1.3.26)
+AC_PROG_SWIG(2.0.0)
 test ! -x "$SWIG" && SWIG=""
 AC_ARG_WITH([swig],
   [AS_HELP_STRING([--with-swig], [Use swig to generate qmf bindings.])],
