Index: qpid/cpp/src/qpid/sys/Shlib.h
===================================================================
--- a/qpid/cpp/src/qpid/sys/Shlib.h	(revision 1245201)
+++ b/qpid/cpp/src/qpid/sys/Shlib.h	(working copy)
@@ -23,6 +23,7 @@
  */
 
 #include "qpid/CommonImportExport.h" 
+#include "qpid/sys/IntegerTypes.h" 
 #include <boost/noncopyable.hpp>
 #include <iostream>
 
Index: qpid/cpp/src/qpid/sys/posix/SocketAddress.cpp
===================================================================
--- a/qpid/cpp/src/qpid/sys/posix/SocketAddress.cpp	2011-08-17 18:07:36.000000000 -0400
+++ b/qpid/cpp/src/qpid/sys/posix/SocketAddress.cpp	2012-02-16 16:30:15.311869726 -0500
@@ -84,8 +84,8 @@
 uint16_t SocketAddress::getPort(::sockaddr const * const addr)
 {
     switch (addr->sa_family) {
-        case AF_INET: return ntohs(((::sockaddr_in*)addr)->sin_port);
-        case AF_INET6: return ntohs(((::sockaddr_in6*)addr)->sin6_port);
+        case AF_INET: return ntohs(((const ::sockaddr_in*)addr)->sin_port);
+        case AF_INET6: return ntohs(((const ::sockaddr_in6*)addr)->sin6_port);
         default:throw Exception(QPID_MSG("Unexpected socket type"));
     }
 }
Index: qpid/cpp/src/qpid/sys/posix/Socket.cpp
===================================================================
--- a/qpid/cpp/src/qpid/sys/posix/Socket.cpp	(revision 1245117)
+++ b/qpid/cpp/src/qpid/sys/posix/Socket.cpp	(working copy)
@@ -29,6 +29,7 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/errno.h>
+#include <unistd.h>
 #include <netinet/in.h>
 #include <netinet/tcp.h>
 #include <netdb.h>
